VAMobile/src/screens/HealthScreen/Pharmacy/PrescriptionDetails/PrescriptionDetails.tsx

```diff
@@ -117,9 +103,13 @@ function PrescriptionDetails({ route, navigation }: PrescriptionDetailsProps) {
               // Call refill request so its starts the loading screen and then go to the modal
               if (!prescriptionInDowntime) {
                 logAnalyticsEvent(Events.vama_rx_request_confirm(prescriptionIds))
-                dispatch(requestRefills([prescription]))
+                const mutateOptions: MutateOptions<RefillRequestSummaryItems, Error, PrescriptionsList, void> = {
+                  onSettled(data) {
+                    navigateTo('RefillScreenModal', data)
+                  },
```

Comment:
Is this supposed to navigate to `RefillScreenModal` when it's a success and failure, or just when it's successful?

Reply:
it's supposed to be on success and failure. What happens after requesting refills is that it gets navigated to the refill screen modal if it wasn't already there(on prescription details so they go back to the right screen on navigation) and then navigates to refill summary screen with a list of successful and failed refills